addons/sale/sale.py

```diff
@@ -588,6 +589,15 @@ def action_cancel(self, cr, uid, ids, context=None):
                         _('Cannot cancel this sales order!'),
                         _('First cancel all invoices attached to this sales order.'))
                 inv.signal_workflow('invoice_cancel')
+
+            # Cancel all procurements linked to the sale order
+            for orderline in sale.order_line:
+                procurement_obj.cancel(cr, uid,
+                    procurement_obj.search(cr, uid, [
+                        ('group_id', '=', orderline.procurement_ids.group_id.id)
+                        ], context=context),
```

Comment:
The procurement_ids only contains the id of the procurement related to 'Partner Locations/Customers' location. Therefore I have to search to find the one in 'WH/Stock' location.